frontend/projects/upgrade/src/app/core/feature-flags/feature-flags.service.ts

```diff
@@ -44,6 +47,13 @@ export class FeatureFlagsService {
     pairwise(),
     filter(([prev, curr]) => prev.status !== curr.status)
   );
+  // Observable to check if selectedFeatureFlag is removed from the store
+  isSelectedFeatureFlagRemoved$ = this.store$.pipe(
+    select(selectSelectedFeatureFlag),
+    pairwise(),
+    filter(([prev, curr]) => prev !== null && prev !== undefined && (curr === null || curr === undefined))
```

Comment:
`prev !== null && prev !== undefined && (curr === null || curr === undefined)` is the same as saying `prev && !curr`, this would work the same and be easier to read:

```
isSelectedFeatureFlagRemoved$ = this.store$.pipe(
  select(selectSelectedFeatureFlag),
  pairwise(),
  filter(([prev, curr]) => prev && !curr)
);
```